src/Instrumentation/Slim/src/SlimInstrumentation.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Slim;
+
+use OpenTelemetry\API\Common\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Contrib\Instrumentation\CallableFormatter;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Slim\App;
+use Slim\Interfaces\InvocationStrategyInterface;
+use Slim\Middleware\RoutingMiddleware;
+use Slim\Routing\RouteContext;
+use Throwable;
+
+class SlimInstrumentation
+{
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation('io.opentelemetry.contrib.php.slim');
+        // @var SpanInterface $root
+        $root = null; //todo probably better to store this in context
+        hook(
+            App::class,
+            'handle',
+            static function (App $app, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation, &$root) {
+                $request = $params[0]; // @var ServerRequestInterface $request
+                $root = $instrumentation->tracer()->spanBuilder(sprintf('%s %s', $request->getMethod(), (string) $request->getUri()->getPath()))
+                    ->setAttribute('code.function', $function)
+                    ->setAttribute('code.namespace', $class)
+                    ->setAttribute('code.filepath', $filename)
+                    ->setAttribute('code.lineno', $lineno)
+                    ->setAttribute(TraceAttributes::HTTP_URL, (string) $request->getUri())
+                    ->setAttribute(TraceAttributes::HTTP_METHOD, $request->getMethod())
+                    ->setAttribute(TraceAttributes::HTTP_REQUEST_CONTENT_LENGTH, $request->getHeaderLine('Content-Length'))
+                    ->setAttribute(TraceAttributes::HTTP_SCHEME, $request->getUri()->getScheme())
+                    ->startSpan();
+                Context::storage()->attach($root->storeInContext(Context::getCurrent()));
+            },
+            static function (App $app, array $params, ?ResponseInterface $response, ?Throwable $exception) {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
+                if (!$scope || $scope->context() === Context::getCurrent()) {
```

Comment:
Why in this case `$scope->context() === Context::getCurrent()` we return?

Reply:
I actually do not know, I copied that part from the psr-18 draft PR!

Reply:
@Nevay if you can explain what this does, I can document it if we think it's good practice for future instrumentations...

Reply:
Not needed here. The PSR18 draft doesn't create a span if an invalid argument is provided and attaches an unmodified context instead*, this check prevents modifying the parent span.
(* Always creating a span and only skipping the usage of argument/request related values might be preferable).